pkg/core/resources/model/resource.go

```diff
@@ -469,8 +469,10 @@ func ComputeLabels(r Resource, mode config_core.CpMode, isK8s bool, systemNamesp
 	}
 
 	getMeshOrDefault := func() string {
-		if mesh := r.GetMeta().GetMesh(); mesh != "" {
-			return mesh
+		if r.GetMeta() != nil {
```

Comment:
I think it's a bit error prone that we have to check `r.GetMeta()` to not be nil, it's easy to forget when adding new labels to `ComputeLabels`. 

For example, it's kind of just a coincidence we don't use use `dataplaneManager` on Kubernetes, otherwise we wouldn't get `k8s.kuma.io/namespace` (because it requires to check `r.GetMeta().GetNameExtensions()`)

Probably we should change `ComputeLabel` signature to not require `ResourceMeta` at all, i.e 
```go
func ComputeLabels(rd ResourceTypeDescriptor, namespace, mesh string, mode config_core.CpMode, isK8s bool, systemNamespace string, localZone string)
```

It's not ideal it's that long, but we can refactor this in the future.

WDYT?